mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,418 @@
+            label_map LabelMap: Class that stores index mappings for a MindMeld app.
+                (Eg. Domain2Id, Intent2Id)
+        Returns:
+            preds_multi (List[List[List]]]): 3D array with probability vectors for unsampled
+                queries from multiple classifiers
+        """
+        skf = StratifiedKFold(
+            n_splits=self.n_classifiers, random_state=ACTIVE_LEARNING_RANDOM_SEED
+        )
+        y = [f"{query.domain}|{query.intent}" for query in sampled_queries]
+        fold_sampled_queries = [
+            [sampled_queries[i] for i in fold]
+            for _, fold in skf.split(sampled_queries, y)
+        ]
+        preds_multi = []
+        for fold_i_queries in fold_sampled_queries:
+            preds_multi.append(
+                self._train_single(
+                    fold_i_queries, unsampled_queries, test_queries, label_map
+                )
+            )
+        return preds_multi
+
+    def domain_classifier_fit_eval(
+        self,
+        sampled_queries: List[ProcessedQuery],
+        unsampled_queries: List[ProcessedQuery],
+        test_queries: List[ProcessedQuery],
+        domain2id: Dict,
+    ):
+        """Fit and evaluate the domain classifier.
+        Args:
+            sampled_queries (List[ProcessedQuery]): List of Sampled Queries
+            unsampled_queries (List[ProcessedQuery]): List of Unsampled Queries
+            test_queries (List[ProcessedQuery]): List of Test Queries
+            domain2id (Dict): Dictionary mapping domains to IDs
+
+        Returns:
+            dc_queries_prob_vectors (List[List]): List of probability distributions
+                for unsampled queries
+            dc_eval_test (): Mindmeld evaluation object for the domain classifier
+        """
+        dc = self.nlp.domain_classifier
+        dc.fit(queries=sampled_queries)
+        dc_eval_test = dc.evaluate(queries=test_queries)
+        dc_queries_prob_vectors = MindMeldALClassifier._get_probs(
+            dc, unsampled_queries, domain2id
+        )
+        return dc_queries_prob_vectors, dc_eval_test
+
+    @staticmethod
+    def _update_eval_stats_domain_level(eval_stats: Dict, dc_eval_test):
+        """Update the eval_stats dictionary with evaluation metrics from the domain
+        classifier.
+
+        Args:
+            eval_stats (Dict): Evaluation metrics to be included in accuracies.json
+            dc_eval_test (): Mindmeld evaluation object for the domain classifier
+        """
+        eval_stats["accuracies"]["overall"] = dc_eval_test.get_stats()["stats_overall"][
+            "accuracy"
+        ]
+        logger.info(
+            "Overall Domain-level Accuracy: %s", eval_stats["accuracies"]["overall"]
+        )
+
+    def intent_classifiers_fit_eval(
+        self,
+        sampled_queries: List[ProcessedQuery],
+        unsampled_queries: List[ProcessedQuery],
+        test_queries: List[ProcessedQuery],
+        domain2id: Dict,
```

Comment:
we dont need the whole mapping here right? just the collection of domain strings

Reply:
Good point, renamed this variable to be `domain_list`. Modified the input coming from the caller of the function accordingly